Parser/tokenizer.c

```diff
@@ -489,25 +489,37 @@ static void fp_ungetc(int c, struct tok_state *tok) {
 
 /* Check whether the characters at s start a valid
    UTF-8 sequence. Return the number of characters forming
-   the sequence if yes, 0 if not.  */
+   the sequence if yes, 0 if not.  The special cases match
+   those in stringlib/codecs.h:decode_utf8.
+*/
 static int valid_utf8(const unsigned char* s)
```

Comment:
Sigh. This file is so old it doesn't even follow our style guide. E.g.
```suggestion
static int
valid_utf8(const unsigned char* s)
```